src/cli/multi.rs

```diff
@@ -21,8 +21,11 @@ pub struct CliMulti {
     #[arg(long)]
     check_in: bool,
     /// The file into which the hostname should be written
-    #[arg(long = "hostname", value_name = "path")]
+    #[arg(group = "hostname", long = "hostname", value_name = "path")]
     hostname_file: Option<String>,
+    /// Parse the hostname retrieved from metadata as a short hostname
+    #[arg(long = "short", requires = "hostname")]
+    short_hostname: bool,
```

Comment:
I think I prefer your original idea of having `--short-hostname=/etc/hostname`.  Rather than adding a mode switch that controls the behavior of another option, let's just support `--hostname` and `--short-hostname` as two output switches that operate independently.